File: Sample Code/DJISdkDemo/Demo/AirLink/AirLinkActionsTableViewController.m
//
//  AirLinkActionsTableViewController.m
//  DJISdkDemo
//
//  Copyright © 2015 DJI. All rights reserved.
//
#import <DJISDK/DJISDK.h>
#import "DemoUtility.h"
#import "AirLinkActionsTableViewController.h"
#import "WiFiLinkSSIDViewController.h"
#import "RebootWiFiViewController.h"
#import "SetGetChannelViewController.h"

@implementation AirLinkActionsTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.sectionNames = [NSMutableArray arrayWithArray:@[@"WiFiLink", @"Lightbridge"]];
    [self.items addObject:@[[DemoSettingItem itemWithName:@"Set/Get WiFi SSID" andClass:[WiFiLinkSSIDViewController class]],
                            [DemoSettingItem itemWithName:@"Reboot WiFi" andClass:[RebootWiFiViewController class]]]];
    
    [self.items addObject:@[[DemoSettingItem itemWithName:@"Set/Get Channel" andClass:[SetGetChannelViewController class]]]];
    
    // The AirLink doesn't support firmware version checking and serial number checking.
}

@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/AirLink/AirLinkActionsTableViewController.swift
//
//  AirLinkActionsTableViewController.m
//  DJISdkDemo
//
//  Copyright © 2016 DJI. All rights reserved.
import DJISDK

class AirLinkActionsTableViewController: DemoTableViewController{
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.connectedComponent = ConnectedProductManager.sharedInstance.fetchAirLink()
        self.sectionNames = ["WiFiLink","Lightbridge"]
        let wifiSetting:[DemoSettingItem] = [DemoSettingItem(name: "Set/Get WiFi SSID", andClass: WiFiLinkSSIDViewController.self), DemoSettingItem(name: "Reboot WiFi", andClass: RebootWiFiViewController.self)]
        self.items.append(wifiSetting)
        
        let lbSetting:[DemoSettingItem] = [DemoSettingItem(name: "Set/Get Channel", andClass:SetGetChannelViewController.self)]

        self.items.append(lbSetting)
        // The AirLink doesn't support firmware version checking and serial number checking.
    }
    
}
